### PR TITLE
Fixed the existing functions and added new routes for subscription

### PR DESCRIPTION
**Pull request by janetlinkiruba**
Checklist
---------

JIRA Ticket -

Checklist (Please mark Yes/No/NA for each of the questions below) -

- I have provided clear and detailed explanation of changes made in the proposed PR : No
- I have compiled the code successfully my laptop : No
- I have written examples to demonstrates the usage of the newly implemented APIs : No
- I have incremented the release version : No
- I have run the Intellij Code Analysis on newly added code and fixed all (valid) reported issues : No

OR

- I have taken approval from scrum master or project manager for skipping any of the items above : No

### CODE REVIEW

File: src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java [hunk 1]
```diff
@@ -90,6 +95,9 @@ public static void main(String[] args) throws IOException {
         // Delete subscription (Launch)
         deleteSubscription(subscriptionClient, createdLaunchSubscriptionId);
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java:96]: deleteSubscription - called twice in this test.

File: src/main/java/com/spotinst/sdkjava/model/SpotinstSubscriptionService.java [hunk 2]
```diff
@@ -125,4 +135,80 @@ public static Boolean deleteSubscription(String subscriptionId, String authToken
...
+        // Handle the response.
+        SubscriptionApiResponse subscriptionApiResponse = getCastedResponse(response, SubscriptionApiResponse.class);
+        if (subscriptionApiResponse.getResponse().getStatus().getCode() == HttpStatus.SC_OK) {
+            retVal = subscriptionApiResponse.getResponse().getItems();
+        }
+
+        return retVal;
+    }
+
+    static Boolean updatSubscription(ApiSubscription subscriptionToUpdate, String subscriptionId, String authToken, String account) throws SpotinstHttpException {
```
Comment (anuragsharma-123) [inline src/main/java/com/spotinst/sdkjava/model/SpotinstSubscriptionService.java:175]: updateSubscription - 'e' missing in function name

File: src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java (hunk 1 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/example/elastigroup/aws/ElastigroupUsageExample.java:96]: fixed

File: src/main/java/com/spotinst/sdkjava/model/SpotinstSubscriptionService.java (hunk 2 above)
Comment (janetlinkiruba) [reply src/main/java/com/spotinst/sdkjava/model/SpotinstSubscriptionService.java:175]: fixed

**Review by anuragsharma-123 — APPROVED** [review]

**Pull request merged**